lib/eosknowledgeprivate/ekn-runtime-document-viewer.c

```diff
@@ -0,0 +1,374 @@
+      !gtk_widget_get_visible (GTK_WIDGET (dialog)))
+    {
+      /* Presenting the window at this point of the webview load avoids showing
+       * a blank webview for a split second!
+       */
+      gtk_window_present (GTK_WINDOW (dialog));
+    }
+}
+
+static void
+ekn_runtime_document_viewer_class_init (EknRuntimeDocumentViewerClass *klass)
+{
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+  GtkWidgetClass *widget_class = GTK_WIDGET_CLASS (klass);
+
+  object_class->finalize = ekn_runtime_document_viewer_finalize;
+  object_class->set_property = ekn_runtime_document_viewer_set_property;
+  object_class->get_property = ekn_runtime_document_viewer_get_property;
+
+  /* Properties */
+  properties[PROP_WEBVIEW] =
+    g_param_spec_object ("webview",
+                         "Web View",
+                         "The WebkitWebView internal child",
+                         WEBKIT_TYPE_WEB_VIEW,
+                         G_PARAM_READABLE);
+
+  properties[PROP_INDEX_URI] =
+    g_param_spec_string ("index-uri",
+                         "Index uri",
+                         "The document index URI to display",
+                         NULL,
+                         G_PARAM_READWRITE);
+
+  properties[PROP_SHOW_ON_LOAD] =
+    g_param_spec_boolean ("show-on-load",
+                          "Shown on Load",
+                          "Defines whether or not the dialog will be shown automaticaly on index_uri load",
+                          TRUE,
+                          G_PARAM_READWRITE);
+
+  g_object_class_install_properties (object_class, N_PROPERTIES, properties);
+
+  /* Template */
+  gtk_widget_class_set_template_from_resource (widget_class, "/com/endlessm/knowledge-private/EknRuntimeDocumentViewer.ui");
+
+  gtk_widget_class_bind_template_child_private (widget_class, EknRuntimeDocumentViewer, webview);
+  gtk_widget_class_bind_template_child_private (widget_class, EknRuntimeDocumentViewer, headerbar);
+  gtk_widget_class_bind_template_child_private (widget_class, EknRuntimeDocumentViewer, back);
+  gtk_widget_class_bind_template_child_private (widget_class, EknRuntimeDocumentViewer, forward);
+
+  gtk_widget_class_bind_template_callback (widget_class, on_back_clicked);
+  gtk_widget_class_bind_template_callback (widget_class, on_forward_clicked);
+  gtk_widget_class_bind_template_callback (widget_class, on_home_clicked);
+  gtk_widget_class_bind_template_callback (widget_class, on_webview_decide_policy);
+  gtk_widget_class_bind_template_callback (widget_class, on_webview_load_changed);
+}
+
+/********************************* Public API *********************************/
+
+/**
+ * ekn_runtime_document_viewer_get_webview:
+ * @dialog: a #EknRuntimeDocumentViewer
+ *
+ * Returns the Webkit web view used to display the URI
+ *
+ * Returns: (transfer none): the webview is owned by the dialog.
+ */
+WebKitWebView *
+ekn_runtime_document_viewer_get_webview (EknRuntimeDocumentViewer *dialog)
```

Comment:
Sorry for forgetting to mention this before. Can you remove this property, I think it's not needed anymore?